### PR TITLE
Save socket.io missing messages during reconnection

### PR DESCRIPTION
**Pull request by starwarfan**
cherry pick #385

### CODE REVIEW

File: doc/Client-Portal Protocol.md [hunk 1]
```diff
@@ -90,7 +91,19 @@ This a format for client reconnects.
 
 **RequestName**: "relogin"<br>
 
-**RequestData**: The ReconnectionTicket object defined in 3.3.1 section.<br>
+**RequestData**: The ReconnectionTicket object defined in 3.3.1 section and PendingMessages:
+
+  object(ReloginResponse)::
+    {
+      ticket: string(Base64Encoded(object(ReconnectionTicket)))
+      messages: [
```
Comment (jianjunz) [inline doc/Client-Portal Protocol.md:99]: Please describe `messages`, at least let developers who don't use our client SDKs know what kind of messages should be sent to server.

File: doc/Client-Portal Protocol.md [hunk 2]
```diff
@@ -90,7 +91,19 @@ This a format for client reconnects.
 
 **RequestName**: "relogin"<br>
 
-**RequestData**: The ReconnectionTicket object defined in 3.3.1 section.<br>
+**RequestData**: The ReconnectionTicket object defined in 3.3.1 section and PendingMessages:
```
Comment (jianjunz) [inline doc/Client-Portal Protocol.md:94]: Please add ``` before and after code to make it better formatted in generated page.

File: doc/Client-Portal Protocol.md [hunk 3]
```diff
@@ -90,7 +91,19 @@ This a format for client reconnects.
...
+
+  object(ReloginResponse)::
+    {
+      ticket: string(Base64Encoded(object(ReconnectionTicket)))
+      messages: [
+        {
```
Comment (jianjunz) [inline doc/Client-Portal Protocol.md:100]: If there is no messages, could it be `undefined` or must be an empty array?

File: doc/Client-Portal Protocol.md [hunk 4]
```diff
@@ -90,7 +91,19 @@ This a format for client reconnects.
...
+          event: string(NotificationName),
+          data: object(NotificationData),
+          seq: number(InternalSeqNo)
```
Comment (jianjunz) [inline doc/Client-Portal Protocol.md:103]: Please define the range of sequence numbers to avoid overflow.

**Comment by jianjunz** [comment]
Will it be a problem if a message is sent twice(before and after reconnection)?

File: doc/Client-Portal Protocol.md (hunk 3 above)
Comment (starwarfan) [reply doc/Client-Portal Protocol.md:101]: no "|| undefined" after "[ ]" means empty array only.

File: doc/Client-Portal Protocol.md (hunk 1 above)
Comment (starwarfan) [reply doc/Client-Portal Protocol.md:100]: Refer to the "NotificationName" and "NotificationData" in this document

**Pull request merged**